### DISCUSSION TITLE
Geospatial function

### DISCUSSION THREAD

**Discussion by davidjianu39** [ibis-project/ibis#3259] [Q&A] [answered]
Hello!

I was wondering if is there any function in Ibis which converts a string into a polygon/point? something similar to ST_ GeomFromText() in SQL? Thank you very much in advance!

**Comment by cpcloud** [comment]
@davidjianu39 It looks like there's no way to do this with ibis right now. We are using `ST_GeomFromText` to construct literal values from `shapely` types, but it appears we do not have an API to convert columns or literals of strings to geometries or geographies.

Is this something you'd be interested in implementing?

**Comment by rcutler-ursa** [comment]
Along these lines, how does one create a literal geometry object that one can perform spatial queries with? For example, 

`data = data.filter(data["geometry"].overlaps(bounding_box))`

where `bounding_box` is a polygon *not* in the database, but rather provided as a constant in the program. Something like: 

`bounding_box = ibis.literal("POLYGON((-71 43, -72 43, -72 44, -71 44, -71 43))", type="geometry")`

**Comment by cpcloud** [comment, marked as answer]
You can construct geometry literals using shapely types:

```
In [9]: import ibis

In [10]: import shapely

In [11]: ibis.literal(shapely.geometry.Point(0, 0))
Out[11]: value: point = <shapely.geometry.point.Point object at 0x7f1a824e0820>

In [12]: ibis.literal(shapely.geometry.Point(0, 0)).type()
Out[12]: Point(nullable=True)
```

and you can also use `tuple`s in the case of `point` and similarly for `polygon`

```
In [25]: ibis.literal(((0, 1), (1, 0), (0, 3)), type="polygon")
Out[25]: value: polygon = ((0, 1), (1, 0), (0, 3))

In [26]: ibis.literal(((0, 1), (1, 0), (0, 3)), type="polygon").type()
Out[26]: Polygon(nullable=True)
```

**Reply by rcutler-ursa** [reply]
Thanks!  Building a `shapely.geometry.Polygon` worked.  Appreciate the quick response.
